src/oop_ext/foundation/types_.py

```diff
@@ -87,47 +90,46 @@ def __getattr__(self, mname):
 
         return self
 
-    def __setattr__(self, _name, _value):
+    def __setattr__(self, _name: str, _value: object) -> Any:
         "Ignore attribute setting."
         return self
 
-    def __delattr__(self, _name):
+    def __delattr__(self, _name: str) -> None:
         "Ignore deleting attributes."
-        return self
 
-    def __enter__(self):
+    def __enter__(self) -> Any:
         return self
 
-    def __exit__(self, *args, **kwargs):
-        return self
+    def __exit__(self, *args: object, **kwargs: object) -> None:
+        pass
 
-    def __repr__(self):
+    def __repr__(self) -> str:
         "Return a string representation."
         return "<Null>"
 
-    def __str__(self):
+    def __str__(self) -> str:
         "Convert to a string and return it."
         return "Null"
 
-    def __bool__(self):
+    def __bool__(self) -> "Literal[False]":
         "Null objects are always false"
         return False
 
     # iter
 
-    def __iter__(self):
+    def __iter__(self) -> Iterator[Any]:
         "I will stop it in the first iteration"
         return self
 
-    def __next__(self):
+    def __next__(self) -> NoReturn:
         "Stop the iteration right now"
         raise StopIteration()
 
-    def __eq__(self, o):
+    def __eq__(self, o) -> Any:
```

Comment:
```suggestion
    def __eq__(self, o) -> bool:
```

Reply:
Actually `Any` is correct here. `__eq__` might return anything the implementation wants, not necessarily `bool`. For example, `==` on numpy arrays results in a another array of `0` and `1`, containing the element wise comparison. 😉